shared/components/map.js

```diff
@@ -59,11 +70,36 @@ export default class Map extends Component {
     Actions.cameraView();
   }
 
+  updatePrevValues() {
+    this.prevLat = this.props.currentLocation.latitude;
+    this.prevLng = this.props.currentLocation.longitude;
+    this.prevZoom = Math.round(this.props.zoomLevel * 100) / 100;
+  }
+
   onLocationChange(region) {
-    this.props.getZoomLevel(this.props.currentLocation.latitudeDelta);
+    const needToFetch = () => {
+      if (!this.prevZoom || this.prevZoom !== Math.round(this.props.zoomLevel * 100) / 100) {
+        return true;
+      }
+      if (Math.abs(this.prevLat - this.props.currentLocation.latitude) >
+        this.props.currentLocation.latitudeDelta) {

```

Comment:
Did you test it? the delta values does not correspond to the lat long bounds. I think it would malfunction at some point.